### DISCUSSION TITLE
Revalidating tags that contain identifiers in their name + `unstable_cache`

### DISCUSSION THREAD

**Discussion by mikevercoelen** [vercel/next.js#63801] [Help]
### Summary

Super quick question:

- In various examples we see cache tags based on a collection i.e. `products`
- This seems to me ineffecient because lets say a user mutates a single product, and I have 1000s warm laying in my cache, in the server action that mutates, we would call i.e. `revalidateTag('products')` which then purges the cache for ALL products.
- What I really want is to JUST invalidate that specific product.

Is it even possible / recommended to use a tag that contains an identifier? Because, if we have this:

```ts
const getProduct = unstable_cache(async (productId) => 
   // Database call
}, ['products'])
```

How could you even dynamically build the cache tag, since it's based on an argument of the function?

I suppose we could do this, but I'm not sure if there are downsides:

```ts
const getProduct = (productId: number) => unstable_cache(async () => {
 // Database call
 }, [`products-${productId}`]));
 ```

**Comment by markedwards** [comment]
I too am curious if this is a wise strategy, or if there is a better approach.
